### DISCUSSION TITLE
When my file contains GIT-DONT-COMMIT, Git should know not to commit it.

### DISCUSSION THREAD

**Discussion by racrawford-fadv** [community/community#108556] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

General

### Body

From time to time I make changes to files that I just want to stay local.
And then later I forget, accidentally checking those files in.
If I put GIT-DONT-COMMIT or GIT-DO-NOT-COMMIT anywhere in my file's contents, Git should refuse to stage or commit that file, and complain to me for trying.

```typescript
// GIT-DONT-COMMIT
const secretAdminPassword = 'P@$5w0rd!';
```

**Comment by thanoulis** [comment]
You should try [gitignore](https://git-scm.com/docs/gitignore).

**Reply by ghost** [reply]
gitignore can exclude based on file pathes, not file contents. You should use [pre-commit](https://git-scm.com/docs/githooks#_pre_commit) hook, which can be any executable script that will be executed before every commit, when it retuns exit code 0 commit will be done, non-zero exit code will abort commit - you could write a script that searches for this terms and also prints a message "do no commit encoutered" before returning non-zero status code

**Comment by ibaddKhan** [comment]
A file is created named .gitignore where you type the names of files  using .fileName and folders by their name only.
those files or folder will remain untracked on git/github
